ha/core/controllers/pcs/cluster_controller.py

```diff
@@ -241,4 +247,53 @@ def add_storageset(self, nodeid: str = None, descfile: str = None) -> dict:
             ([dict]): Return dictionary. {"status": "", "msg":""}
                 status: Succeeded, Failed, InProgress
         """
-        raise HAUnimplemented("This operation is not implemented.")
\ No newline at end of file
+        raise HAUnimplemented("This operation is not implemented.")
+
+    @controller_error_handler
+    def create_cluster(self, name: str, user: str, secret: str, nodeid: str) -> dict:
+        """
+        Create cluster if not created.
+
+        Args:
+            name (str): Cluster name.
+            user (str): Cluster User.
+            secret (str): Cluster passward.
+            nodeid (str): Node name, nodeid of current node.
+
+        Returns:
+            dict: Return dictionary. {"status": "", "msg":""}
+        """
+        status = ""
+        try:
+            self._check_non_empty(name=name, user=user, secret=secret, nodeid=nodeid)
+            if not self._is_pcs_cluster_running():
+                self._auth_node(nodeid, user, secret)
+                self._execute.run_cmd(const.PCS_SETUP_CLUSTER.replace("<cluster_name>", name)
+                                        .replace("<node>", nodeid), is_secret=True,
+                                        error="Cluster setup failed.")
+                Log.info("Pacmaker cluster created, waiting to start node.")
+                self._execute.run_cmd(const.PCS_CLUSTER_START_NODE)
+                self._execute.run_cmd(const.PCS_CLUSTER_ENABLE.replace("<node>", nodeid))
+                Log.info("Node started and enabled successfully.")
+                # TODO: Divide class into vm, hw when stonith is needed.
+                self._execute.run_cmd(const.PCS_STONITH_DISABLE)
+                time.sleep(const.BASE_WAIT_TIME)
+
+            # Check node status
+            node_status = self.nodes_status([nodeid]).get(nodeid).lower()
```

Comment:
"node standby" functionality should not be part of create cluster. Please move this code to node controller.

Reply:
Done